src/lib.rs

```diff
@@ -1201,13 +1200,14 @@ pub unsafe trait TryFromBytes {
 
     /// Attempts to interpret the prefix of the given `candidate` as a `&Self`
     /// without copying.
-    ///
-    /// If the first `size_of::<Self>()` bytes of `candidate` are a valid
-    /// instance of `Self`, this method returns both a reference to those bytes
-    /// interpreted as a `Self`, and a reference to the remaining bytes. If
-    /// `candidate.len() < size_of::<Self>()` or `candidate` is not aligned to
-    /// `align_of::<Self>()` or the bytes are not a valid instance of `Self`,
-    /// this returns `Err`.
+    /// 
+    /// This method attempts to deserialize the largest possible size of `Self`
+    /// from the leading bytes of `candidate`. If that prefix is a valid
+    /// instance of `Self`, this method returns a reference to those bytes
+    /// reinterpreted as `Self`, and a reference to the remaining bytes. If
+    /// there are insufficient bytes, or if `candidate` is not appropriately
```

Comment:
```suggestion
    /// there are insufficient bytes, if `candidate` is not appropriately
```

Reply:
I think the "or" is load-bearing. We're describing a disjunction of error conditions, not a conjunction.

Reply:
"If x, y, or z" describes a disjunction, just like "if x, y, and z" describes a conjunction. Am I missing something?

Reply:
The structure you're proposing isn't "if x, y, or z", it's "If x, if y, or z". The operator precedence isn't totally unambiguous.

And if we used "if x, y, or z", it's not clear until that very last "or" whether the reader should understand those conditions to be in a conjunction or disjunction. We shouldn't hold the reader in suspense like that.

Reply:
> And if we used "if x, y, or z", it's not clear until that very last "or" whether the reader should understand those conditions to be in a conjunction or disjunction. We shouldn't hold the reader in suspense like that.

I'm convinced by this.